toolkit/src/cli/data_requests.rs

```diff
@@ -27,8 +36,29 @@ pub(crate) fn decode_from_args(
 pub(crate) fn try_from_args(
     args: arguments::TryDataRequest,
 ) -> Result<RADRequestExecutionReport, Error> {
-    let request = decode_from_args(arguments::DecodeDataRequest { hex: args.hex })?.data_request;
     let full_trace = args.full_trace.unwrap_or(true);
+    let request = decode_from_args(args.into())?.data_request;
 
     lib::data_requests::try_data_request(&request, full_trace)
 }
+
+/// Extract the Protocol Buffers representation of a data request from a Solidity smart contract
+/// that is an instance of the `WitnetRequest.sol` contract, or at least implements the same
+/// interface.
+fn extract_pb_bytes_from_solidity_file(path: &Path) -> Result<Vec<u8>, Error> {
+    let mut file = File::open(path).map_err(Error::SolidityFileCantOpen)?;
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)
+        .map_err(Error::SolidityFileCantRead)?;
+
+    let hex_reg_ex: Regex =
+        Regex::new(r#"\s*constructor.*public\s*Request\s*\(\s*hex"([\da-f]+)"#)?;
```

Comment:
This regex does not match the example request from here: https://docs.witnet.io/tutorials/bitcoin-price-feed/compiling/

Reply:
True. `public` is not used in constructors anymore. I adjusted the regex to account for either presence or absence of such modifier.

Reply:
Now it does match the request but the bytes are not valid protobuf. So we should probably update that tutorial.

Reply:
Yep, it seems those are totally outdated or otherwise simply broken.